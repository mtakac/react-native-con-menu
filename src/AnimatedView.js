import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Animated } from 'react-native';

import withDelayedUnmout from './withDelayedUnmount';

const propTypes = {
    isMounted: PropTypes.bool,
    duration: PropTypes.number.isRequired,

    style: PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.number
    ])
};

const defaultProps = {
    isMounted: false,
    style: null
};

class AnimatedView extends Component {
    state = { opacity: new Animated.Value(0) }

    componentDidMount() {
        const { duration } = this.props;
        const { opacity } = this.state;

        Animated.timing(opacity, { toValue: 1, duration }).start();
    }

    componentDidUpdate(prevProps) {
        const { isMounted, duration } = this.props;
        const { opacity } = this.state;

        if (prevProps.isMounted && !isMounted) {
            Animated.timing(opacity, { toValue: 0, duration }).start();
        }
    }

    render() {
        const { style, children } = this.props;
        let { opacity } = this.state;

      return <Animated.View style={{ ...style, opacity: opacity }}>{children}</Animated.View>;
    }
}

AnimatedView.propTypes = propTypes;
AnimatedView.defaultProps = defaultProps;

export default withDelayedUnmout(AnimatedView);
