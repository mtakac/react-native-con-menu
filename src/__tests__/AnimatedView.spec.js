import React from 'react';
import { shallow } from 'enzyme';
import AnimatedView from '../AnimatedView';

describe('<AnimatedView />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<AnimatedView duration={500} />);
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        wrapper.setProps({ isMounted: true });
        expect(wrapper).toMatchSnapshot();
    });
});
