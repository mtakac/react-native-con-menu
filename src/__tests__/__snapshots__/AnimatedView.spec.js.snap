// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedView /> mounts correctly 1`] = `
<AnimatedView
  duration={500}
  isMounted={true}
  style={null}
/>
`;

exports[`<AnimatedView /> renders correctly 1`] = `""`;
